### PR TITLE
New GOST PKCS12 standard support

### PR DESCRIPTION
**Pull request by beldmit**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by beldmit** [comment]
Travis finally got happy. Ping @openssl/otc for the review.

File: crypto/pkcs12/p12_decr.c [hunk 1]
```diff
@@ -24,22 +24,52 @@ unsigned char *PKCS12_pbe_crypt(const X509_ALGOR *algor,
...
+            PKCS12err(PKCS12_F_PKCS12_PBE_CRYPT, ERR_R_INTERNAL_ERROR);
+            goto err;
+        }
+
+        if (EVP_CIPHER_CTX_encrypting(ctx)) {
+            max_out_len += mac_len;
+        } else {
+            if (inlen < mac_len) {
+                PKCS12err(PKCS12_F_PKCS12_PBE_CRYPT,
+                PKCS12_R_UNSUPPORTED_PKCS12_MODE);
```
Comment (paulidale) [inline crypto/pkcs12/p12_decr.c:60]: Indentation is odd here.

**Review by paulidale — APPROVED** [review]
Look okay.

File: crypto/pkcs12/p12_decr.c (hunk 1 above)
Comment (beldmit) [reply crypto/pkcs12/p12_decr.c:60]: Hmmm... It's indented with the `if` clause corresponding to `else`. Is it wrong?

Comment (beldmit) [reply crypto/pkcs12/p12_decr.c:60]: Oh, got your point! Thanks!

Comment (beldmit) [reply crypto/pkcs12/p12_decr.c]: Fixup pushed.

**Comment by beldmit** [comment]
@paulidale, if there are no other problems, could you please reapprove and add the corresponding label?

**Comment by slontis** [comment]
@paulidale ?

**Review by paulidale — APPROVED** [review]

**Comment by beldmit** [comment]
Merged. Thanks!

**Pull request closed without merging**